### PR TITLE
Mobile click accessibility

### LINKED ISSUE #8178: VoiceOver unable to interact with buttons
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem
The hamburger/menu button has 2 major issues:

1. It's read as "Button" - no label
2. Activating it does not work

## Expected behavior

With VoiceOver enabled, I can interact with the menu button.

## Steps to reproduce

1. Turn on VoiceOver: Settings, Accessibility, VoiceOver, enable.
1. Navigate to the Menu button in the Home Assistant app or in Safari
2. Note the label read: "Button"
3. Double tap to activate the menu button
4. Note the menu does not open

## Environment

- Home Assistant release with the issue: 2021.1.3
- Last working Home Assistant release (if known): none known
- Browser and browser version: iOS 14.3's Safari
- Operating system: iOS 14.3

## State of relevant entities
n/a

## Problem-relevant frontend configuration

Occurs on a vanilla install.

## Additional information

Message me up on Discord and I can walk you through testing VoiceOver in the iOS simulator or on a physical device.

### LINKED ISSUE #10617: Accessibility regression: cannot click buttons using Home Assistant companion for Android with screen reader
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Using Android Home Assistant app with TalkBack, when going to configuration then server control, I cannot click any button, for example restart or check config.
Moreover, I have every button twice and none can be used.
Using Chrome on a desktop I still have buttons twice but the second one is recognized as a button and can be clicked.

### Describe the behavior you expected

I should be able to double-tap the button to activate it.

### Steps to reproduce the issue

1. Open Android App with TalkBack enabled by placing your finger on Home Assistant then double-tapping
2. Activate side bare by placing the finger on then double-tapping the button to show it
3. Place a finger then double-tap configuration
4. Place the finger to server control then double-tap
5. Put your finger on a button, then double-tap

### What version of Home Assistant Core has the issue?

2021.11

### What was the last working version of Home Assistant Core?

2021.10

### In which browser are you experiencing the issue with?

Home Assistant android app

### Which operating system are you using to run this browser?

Android 12

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

It is a really annoying accessibility regression which make unusable some elements of Home Assistant interface, it looks like others buttons are no more recognized by TalkBack

### LINKED ISSUE #11034: Can't activate links or navigation bar items using VoiceOver in web UI.
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When using VoiceOver, the screen reader for people who are blind on IOS and Mac OS devices, it is not possible to activate links in the navigation menu or the configuration menu.  

It seems that this is because it expects the enter key to be pressed in order for the link to be activated. Tapping on the item sets focus to it.  I'm informed that visually, this results in the expected focus colour change but it doesn't let me move to that page. 

Please keep in mind that this UI works beautifully on Windows.   Aside from a few niggling things that are quite minor, I really have no complaints when using HomeAssistant in Windows. 

If you are interested, I am blind.  I'm a senior systen engineer and have worked with technology to pay the bills for 20 years. I only started using HomeAssistant about 2 months ago.  Since then I have spent *way too much* on smarg home tech and I have loved every minute of working on it.  I'm thrilled by how accessible HomeAssistant has been right out of the box and I have actually Emailed the maintainers to highlight this.  Well done to you all for doing such an amazing job.  If I can be of any assistance to you please let me know. 

### Describe the behavior you expected

When you double tap on a navigation item, the associated page should open. 

### Steps to reproduce the issue

1.  Open a browser in IOS. It doesn't matter what one. 
2.  Open the homeAssistant web UI. 
3.  Try to move to one of the items in the navigation menu. 
4.   If you are in a dashboard, you will notice that you can toggle buttons etc. But the navigation menu doesn't work as expected. 

Expected result.  When you double tap on a navigation item, the associated page should open. 

### What version of Home Assistant Core has the issue?

2021.12.6

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Safari

### Which operating system are you using to run this browser?

IOS 15 

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

Change the Polymer setting `cancelSyntheticClickEvents` to false globally in the UI.  When left to true, it ends up breaking nearly all buttons in the UI for mobile screen readers.

_PS - I know the 2nd commit is only loosely related, but that comment would have saved me a bunch of time trying to figure out why putting the code there wasn't working_

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

N/A - Just flip VoiceOver or Talkback on with and without this change.

## Additional information

- This PR fixes or closes issue: fixes #8178, fixes #10617
- This PR is related to issue or discussion: Polymer/polymer#5697

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

**Comment by steverep** [comment]
Sorry, forgot to mention that I have no custom panel to test, but the 2 lines of code are identical for each entrypoint.

Also, I assume there is no need to add it to the `core.ts` entrypoint as no UI elements go through there?  And did I miss any edge points that I may not have thought to test?

**Comment by bramkragten** [comment]
Thanks for this PR! I'll dive into this tomorrow

**Comment by steverep** [comment]
This should also close #11034 which seems like a duplicate issue.

**Comment by steverep** [comment]
Any chance this can get in the upcoming Feb release? 🥺

**Comment by bramkragten** [comment]
> Any chance this can get in the upcoming Feb release? 🥺

I don't think that is wise, I would like it to go through a complete beta and some testing before shipping this into production. We are not sure if this will break anything...

**Comment by steverep** [comment]
> I don't think that is wise, I would like it to go through a complete beta and some testing before shipping this into production. We are not sure if this will break anything.

I understand the hesitation on a global setting, but...
- It's already broken!  This is absolutely debilitating to any screen reader user.  There is no work around.  The mobile apps continue to be useless without this fix.
- The vast majority of components, including all `mwc-*` and nearly all `paper-*`, don't use the problematic module where this setting is employed.  The only reason they are affected is because of reckless and intrusive coding IMHO.  See the polymer issue I opened.
- The only documentation I could find on this setting is the single comment where it is declared, and that more or less acknowledges it is not needed and has detrimental effects:
```
* Setting to cancel synthetic click events fired by older mobile browsers. Modern browsers
 * no longer fire synthetic click events, and the cancellation behavior can interfere
 * when programmatically clicking on elements.
 */
export let cancelSyntheticClickEvents = true;
```


If you still think extensive testing is necessary, here are the only 9 modules in the entire source tree that actually import the offending `gestures.js`.  It would not be difficult to quickly test where they are actually used in the frontend (if at all).

- node_modules/@polymer/iron-overlay-behavior/iron-overlay-manager.js:
  - 16: import * as gestures from '@polymer/polymer/lib/utils/gestures.js';
- node_modules/@polymer/paper-dropdown-menu/paper-dropdown-menu-light.js:
  - 27: import * as gestures from '@polymer/polymer/lib/utils/gestures.js';
- node_modules/@polymer/paper-dropdown-menu/paper-dropdown-menu.js:
  - 28: import * as gestures from '@polymer/polymer/lib/utils/gestures.js';
- node_modules/@polymer/paper-slider/paper-slider.js:
  - 21: import {setTouchAction} from '@polymer/polymer/lib/utils/gestures.js';
- node_modules/@polymer/polymer/lib/legacy/legacy-element-mixin.js:
  - 18: import { setTouchAction } from '../utils/gestures.js';
- node_modules/@polymer/polymer/lib/mixins/gesture-event-listeners.js:
  - 13: import { addListener, removeListener } from '../utils/gestures.js';
- node_modules/@vaadin/vaadin-button/src/vaadin-button.js:
  - 8: import { addListener } from '@polymer/polymer/lib/utils/gestures.js';
- node_modules/@vaadin/vaadin-date-picker/src/vaadin-date-picker-mixin.js:
  - 9: import { addListener } from '@polymer/polymer/lib/utils/gestures.js';
- node_modules/@vaadin/vaadin-date-picker/src/vaadin-date-picker-overlay-content.js:
  - 8: import { addListener, setTouchAction } from '@polymer/polymer/lib/utils/gestures.js';

**Comment by bramkragten** [comment]
I'm not worried about issues with current browsers but with the old browsers, like mentioned in that comment.

I'm not sure what devices/browsers are affected.

I guess we can put this in, as the beta generally doesn't find issues with old browsers anyway...

File: src/entrypoints/custom-panel.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // Compat needs to be first import
 import "../resources/compatibility";
+import { setCancelSyntheticClickEvents } from "@polymer/polymer/lib/utils/settings";
```
Comment (bramkragten) [inline src/entrypoints/custom-panel.ts:3]: We don't need it here, as we don't load Polymer code ourself

**Review by bramkragten — APPROVED** [review]

File: src/entrypoints/custom-panel.ts (hunk 1 above)
Comment (steverep) [reply src/entrypoints/custom-panel.ts:3]: I wasn't sure about putting it there, but I assumed it was needed since polymer was imported a few lines down.

**Comment by bramkragten** [comment]
`paper-dropdown-menu` seems not to work anymore on iOS, I propose we revert this PR for now, and work on replacing all `paper-dropdown-menu` with `mwc-select`. After that we can enable this again (next release?).

**Comment by bramkragten** [comment]
In the meantime if needed, people could just stay on version 2022.2.0 until we have fixed this.

**Pull request merged**